ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -1,6 +1,22 @@
-//! The Yk JIT IR
+//! The JIT's Intermediate Representation (IR).
 //!
-//! This is the in-memory trace IR constructed by the trace builder and mutated by optimisations.
+//! This is the IR created by [trace_builder] and which is then optimised. The IR can feel a little
+//! odd at first because while it fundamentally represents a graph, the in-memory representation is
```

Comment:
Why is it a graph again? It's not to do with control flow, is it? Is it because constants can be shared by multiple instructions?

Reply:
> It's not to do with control flow, is it?

There is an element of this, because of looping, but in practise mostly:

> Is it because constants can be shared by multiple instructions?

Yes, mostly due to this sort of thing.

Reply:
I wonder if it might be worth clarifying, maybe giving an example? If I wasn't sure, then newcomers won't be sure.

Reply:
Slightly changed the wording in 8c9f71bc9b47e1046de268d5710bf3a400fdc557 to focus on the most important parts.